vms/proposervm/batched_vm.go

```diff
@@ -97,9 +94,11 @@ func (vm *VM) BatchedParseBlock(ctx context.Context, blks [][]byte) ([]snowman.B
 		statelessBlockDescs = make([]partialData, 0, len(blks))
 		innerBlockBytes     = make([][]byte, 0, len(blks))
 	)
+
+	parsingResults := vm.parseBlocks(blks)
+
 	for ; blocksIndex < len(blks); blocksIndex++ {
```

Comment:
That's a good question. I don't know if the caller code ensures that the blocks are passed in-order.

I would guess that if not, then probably we could.

However I wanted to preserve the semantics of this change in this PR.

@StephenButtolph wdyt?

Reply:
@marun I think we want to keep the batched call to the innerVM here, so we are bound to wait for all proposerVM level parsing to be done before pushing the innerBlocks content down to the innerVM.
IIUC, you're suggesting instead to avoid batching innerVM calls and shoot them as soon as proposerVM has done its work on a single block.
I am not sure this is helpful, especially if innerVM is an rpcVM one, which is the case that brought us to introduce batching in the first place.

Reply:
> Do blocks need to be handled by this loop serially or would it be possible (or sane) to optimize this loop further to process in the order that blocks are successfully parsed (e.g. via a channel)?

As this PR is currently written - there is no real change in behavior... It is just an optimization.

The first block bytes provided must remain the first block returned in the slice (this is required by the engine). Technically I don't think anything would break if the rest of the blocks were potentially shuffled... But I feel like that would be a pretty weird API. IMO we should document on the `BatchedParseBlock` interface method (and helper function) that the order is expected to be maintained.

If we were to use a channel here, we would need to remove the:
```golang
if err != nil {
	break
}
```
behavior, as we may still need to process additional blocks.